bugwarrior/config.py

```diff
@@ -184,9 +185,27 @@ def get_taskrc_path(conf, main_section):
     )
 
 
-def get_data_path():
-    return os.path.expanduser(os.getenv('TASKDATA', '~/.task'))
+@lru_cache()
+def get_data_path(config, main_section):
+    taskrc = get_taskrc_path(config, main_section)
 
+    # We cannot use the taskw module here because it doesn't really support
+    # the `_` subcommands properly (`rc:` can't be used for them).
+    line_prefix = 'data.location='
+    tw_show = subprocess.Popen(
+        ('task', '_show'), stdout=subprocess.PIPE, env={'TASKRC': taskrc})
+    data_location = subprocess.check_output(
+        ('grep', line_prefix), stdin=tw_show.stdout)
+    tw_show.wait()
+    data_path = data_location[len(line_prefix):].rstrip()
+
+    if not data_path:
+        data_path = os.getenv('TASKDATA', '~/.task')
+        log.warning(
+            'Unable to determine the data location. Falling back to %s.' %
+            data_path)
```

Comment:
I was thinking the more likely scenario than `data.location` not being set would be that either the `task` binary isn't in the `PATH` or `taskwarrior _show`'s output changes and breaks our regex. Perhaps it would be better to just break in those cases than to screw up the data.